src/nvme/fabrics.c

```diff
@@ -1052,57 +1052,62 @@ nvme_ctrl_t nvmf_connect_disc_entry(nvme_host_t h,
 	return NULL;
 }
 
-static struct nvmf_discovery_log *nvme_discovery_log(nvme_ctrl_t c,
-						     struct nvme_get_log_args *args,
-						     int max_retries)
+/*
+ * Most of nvmf_discovery_log is reserved, so only fetch the initial bytes.
+ * 8 bytes for GENCTR, 8 for NUMREC, and 2 for RECFMT.
+ * Since only multiples of 4 bytes are allowed, round 18 up to 20.
+ */
+#define DISCOVERY_HEADER_LEN 20
+
+static struct nvmf_discovery_log *nvme_discovery_log(
+	const struct nvme_get_discovery_args *args)
 {
-	nvme_root_t r = c->s && c->s->h ? c->s->h->r : NULL;
-	struct nvmf_discovery_log *log = NULL;
-	int ret, retries = 0;
-	const char *name = nvme_ctrl_get_name(c);
+	nvme_root_t r = args->c->s && args->c->s->h ? args->c->s->h->r : NULL;
```

Comment:
You could use the helper function `root_from_ctrl()` as follows:
```
nvme_root_t r = root_from_ctrl(args->c);
```

Reply:
Yup, will do